shared-libs/lineage/src/utils.js

```diff
@@ -0,0 +1,27 @@
+const contactTypeUtils = require('@medic/contact-types-utils');
+const _ = require('lodash/core');
+
+const isContact = doc => {
+  if (!doc) {
+    return;
+  }
+
+  return doc.type === 'contact' || contactTypeUtils.HARDCODED_TYPES.includes(doc.type);
+};
+
+const getId = (item) => item && (typeof item === 'string' ? item : item._id);
+
+// don't process linked docs for non-contact types
+// linked_docs property should be a key-value object
+const validLinkedDocs = doc => {
+  if (!doc || !isContact(doc) || !doc.linked_docs) {
+    return;
+  }
+
+  return _.isObject(doc.linked_docs) && !_.isArray(doc.linked_docs);
```

Comment:
Nit: we're also checking `!doc` in `isContact` so we don't need to do it again. Also, _.isObject returns false for undefined, so I think we don't need that either.

It might also be cleaner as a one liner now, eg:

```suggestion
  return isContact(doc) && _.isObject(doc.linked_docs) && !_.isArray(doc.linked_docs);
```

Reply:
Changed!